### ISSUE TITLE
Segments loaded from DICOM should have names as in DICOM SEG

### ISSUE THREAD

**Issue by fedorov** [QIICR/QuantitativeReporting#119]
Currently segments names in the measurements table are correctly set from DICOM, but the segments list names are incorrect.

![image](https://cloud.githubusercontent.com/assets/313942/20640420/59c38ac0-b3ab-11e6-92ce-7c391d231069.png)

**Comment by che85** [comment]
I am not sure, how the Name column is populated. For the measurements table, I am calling segment.GetName()

@cpinter How are those names populated when loading them from DICOM?

**Comment by fedorov** [comment]
Isn't there a setter for the name?!

**Comment by che85** [comment]
Yes there is, but it doesn't make sense to me that the names are displayed incorrectly. When you create a segmentation on your own and then change the segment name from the segment table, then it will also change in the measurements table. That's why I am wondering.

**Comment by cpinter** [comment]
In the DICOM SEG loader, the segment name is the same as the temporary labelmap volume's name that was generated. I believe those nrrds are written out by your CLI

**Comment by cpinter** [comment]
This is where segment name is set when loading:
https://github.com/QIICR/QuantitativeReporting/blob/master/Py/DICOMSegmentationPlugin.py#L278
I don't know where the proper name should come from, but it needs to be set here

**Comment by che85** [comment]
@cpinter Wouldn't it make sense to populate that from the terminology?

**Comment by che85** [comment]
But in any case, there is something wrong then. I will need to check .

**Comment by fedorov** [comment]
I will need to look as well. As I think about it, there is a name for the measurements group in SR, and segment label in SEG. They don't have to be the same, and I think they are not. Please don't spend any time on this until I take a look. Sorry for the confusion!

**Comment by fedorov** [comment]
In either case, there is a setter for segment name, right Csaba?

**Comment by cpinter** [comment]
@che85 I didn't make any decisions re segment name, I simply made it work that was implemented at the NA-MIC week by Kyle (if I remember correctly).

@fedorov segment->SetName(newName)

**Comment by cpinter** [comment]
Please see the line I referred to a few comments above. The argument of that call simply needs to be replaced from the proper name. I'm not sure what I can do to help further.

**Comment by fedorov** [comment]
Csaba, I will look at it later today or tomorrow. Thanks for the clarification, nothing else is needed.

**Comment by fedorov** [comment]
Ok, so there are several issues:

* segment name is set here https://github.com/QIICR/QuantitativeReporting/blob/master/Py/DICOMSegmentationPlugin.py#L228 as `seriesName + "-" + typeCodeMeaning + "-label"`, which does not make much sense
* segment name should probably be set to DICOM attribute SegmentLabel (Type 1), but that attribute is not included in the schema (but SegmentDescription is, which is Type 3!); currently we set segment label to be the same as SegmentatioCategoryType code meaning https://github.com/QIICR/dcmqi/blob/master/libsrc/ImageSEGConverter.cpp#L213-L215
* measurements groups have "Tracking Identifier" in SR, and it is in the general case different from SegmentLabel in SEG

I think here's what we should do:
- [ ] add an option to specify SegmentLabel to the SEG schema
- [ ] save SegmentLabel into JSON file while converting from SEG
- [ ] while loading segments, assign segment name to be SegmentLabel from JSON
- [ ] while populating the measurements table, have one column in the table to be "Tracking Identifier" loaded from SR (it is in the schema), and another one for the SegmentLabel of the segment that was used to calculate the measurements in that measurements group

**Comment by fedorov** [comment]
To avoid any confusion - none of this is Csaba's fault! :)
